src/plan_expand_hypertable.c

```diff
@@ -70,6 +72,124 @@ is_chunk_exclusion_func(Expr *node)
 	return false;
 }
 
+static bool
+is_time_bucket_function(Expr *node)
+{
+	if (IsA(node, FuncExpr) &&
+		strncmp(get_func_name(castNode(FuncExpr, node)->funcid), "time_bucket", NAMEDATALEN) == 0)
+		return true;
+
+	return false;
+}
+
+/*
+ * Transform time_bucket calls of the following form in WHERE clause:
+ *
+ * time_bucket(width, column) > value
+ * column > value - width
+ *
+ * time_bucket(width, column) < value
+ * column < value + width
+ *
```

Comment:
writing down notes on why this works:
`time_bucket` rounds down, so `column - width < time_bucket(width, column) <= column`
for the first equation, with any non-negative `width`
```
column >= time_bucket(width, column) > value
column > value
column > value - width
```
for the second
```
column - width < time_bucket(width, column)  < value
column - width < value
column < value + width
```
so we might be more conservative than we need to be, but it should work.

Reply:
I think we need more of the justification in this comment.

Reply:
for lower bound comparison the width does not actually matter so i adjusted the logic and also tried documenting it a bit more